### PR TITLE
Background computation of pgx

### PR DESCRIPTION
**Pull request by mauromiguelm**
**Summary:**

- Create an external process that handles PGX computation
- Allow user to mine other datasets while PGX is computed
- Allow multiple dataset being processed at the same time (`MAX_PROCESS_COUNT`). 
- Script `pgxcreate_op.R` based on `playbase` that computes PGX externally and does not require sourcing OP
- add option to run script pgxcreate_op.R from command line
- Disable computing of new datasets if number of external processes > MAX_PROCESS_COUNT.
- External processing works on `Linux`, `Windows` and `Mac`
- Changes in UI/UX to accomodate new features

**Future needs:**

- Move MAX_PROCESS_COUNT as a global parameter (current default to 2)
- Test on docker deployment
- Have a SSO MAX_PROCESS_COUNT
- OPTIONS.yml and its default options

Thanks Nick for feedback and testing on Mac and Linux!

Closes issue #270

### CODE REVIEW

File: bin/pgxcreate_op.R
[the remark this answers is not in the document]
Comment (mauromiguelm) [reply bin/pgxcreate_op.R:1]: Hi Nick, I agree, we can wait to update the script accordingly, but not much change is needed besides renaming 2x functions I believe. 

Playbase changes are out of scope of this PR, so I guess waiting is the solution (?).

[the remark this answers is not in the document]
Comment (mauromiguelm) [reply bin/pgxcreate_op.R:1]: Hi Nick, I got an error with a missing libs file on playbase, what is the right way to add it to the package?

`[pgx.computePGX] testing genesets...
Error in gzfile(file, "rb") : cannot open the connection
Calls: <Anonymous> -> compute.testGenesets -> readRDS -> gzfile
In addition: Warning message:
In gzfile(file, "rb") :
  cannot open compressed file '../lib/gset-sparseG-XL.rds', probable reason 'No such file or directory'
Execution halted`

**Comment by ivokwee** [comment]
What will happen if computation gets error and stops? Do we get an error message?

**Comment by mauromiguelm** [comment]
The error log should be shown by `options(shiny.errors)` to user in the panel that Xavier created. We could addtitionally bring the temporary folder to the user folder, where any PARAM that is not accompanied by pgx will indicate a problem with creating pgx. This way the `PARAMS` with all initial conditions should be easily available for debug.

**Pull request merged**